addOns/network/src/main/java/org/zaproxy/addon/network/internal/client/BaseHttpSender.java

```diff
@@ -92,6 +95,11 @@ protected BaseHttpSender() {
                                 AbstractReferenceMap.WEAK, AbstractReferenceMap.HARD));
     }
 
+    @Override
+    public void setRateLimiter(RateLimiter rateLimiter) {
+        this.rateLimiter = rateLimiter;
```

Comment:
This should have a null check and fallback to `NopRateLimiter`. It would also be better to only set the other implementation when there ar enabled rules, no point doing any work at all if we know there's none (which should be the most common case).